src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -453,7 +453,16 @@ public function run() {
 
     $this->runDiffSetupBasics();
 
-    $commit_message = $this->buildCommitMessage();
+    $revert_plan_check_paths = $this->getConfigurationManager()->getConfigFromAnySource(
+          'differential.revert_plan_check_paths');
+    if (!is_null($revert_plan_check_paths)) {
+      // revert plan checking only happens for certain paths in the repository
+      // so, we need to compute the changes early in this case
+      $changes = $this->generateChanges();
```

Comment:
please verify that `--no-lint` and `--no-unit` below still work as expected with the check path enabled.